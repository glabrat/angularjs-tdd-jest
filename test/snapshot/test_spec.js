import TodoListTemplate from "components/todoList.component.html";

describe("Snapshot Testing on AngularJS", () => {
    const {Â angularjsSnapshotFormat } = require(`${__dirname}/angularjsSnapshotFormat`)
    const todos = [
        {
            "name": "Learn Programming using component based approach",
            "completed": true
        },
        {
            "name": "Learn Machine Learning",
            "completed": false
        }
    ];
    let scope;

    beforeEach(inject(($rootScope) => {
        scope = $rootScope.$new();
    }))

    it("should be render a todo list", () => {
        scope.$ctrl = {
            todosList : todos
        };
        const renderedElement = angularjsSnapshotFormat(TodoListTemplate, scope);

        expect(renderedElement).toMatchSnapshot();
    });
});
