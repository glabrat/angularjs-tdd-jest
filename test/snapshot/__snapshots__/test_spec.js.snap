// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing on AngularJS should be render a todo list 1`] = `
"<div class=\\"todo-list ng-scope\\">
    <div class=\\"todo-item ng-scope\\" ng-repeat=\\"[ngExpression]\\">
        <label class=\\"ng-binding\\">
            <input type=\\"checkbox\\" ng-model=\\"[ngExpression]\\" class=\\"ng-pristine ng-untouched ng-valid ng-not-empty\\"> Learn Programming using component based approach
            <button class=\\"btn-delete-todo\\" ng-click=\\"[ngExpression]\\">x</button>
        </label>
    </div><div class=\\"todo-item ng-scope\\" ng-repeat=\\"[ngExpression]\\">
        <label class=\\"ng-binding\\">
            <input type=\\"checkbox\\" ng-model=\\"[ngExpression]\\" class=\\"ng-pristine ng-untouched ng-valid ng-empty\\"> Learn Machine Learning
            <button class=\\"btn-delete-todo\\" ng-click=\\"[ngExpression]\\">x</button>
        </label>
    </div>
</div>"
`;
